topdown/eval.go

```diff
@@ -2304,3 +2308,13 @@ func refSliceContainsPrefix(sl []ast.Ref, prefix ast.Ref) bool {
 	}
 	return false
 }
+
+func refContainsNonScalar(ref ast.Ref) bool {
+	for _, term := range ref[1:] {
+		switch term.Value.(type) {
```

Comment:
Nit: you can use `ast.IsScalar(term.Value)` here.

Reply:
Doh.. I apparently failed at ctrl-f since I swear I went looking for a helper like that. Will update to use it 😄